lls/main.c

```diff
@@ -719,6 +849,54 @@ register_nd_acl_rules(struct gatekeeper_if *iface)
 	return 0;
 }
 
+static void
+fill_ping6_rule(struct ipv6_acl_rule *rule, struct in6_addr *addr)
+{
+	uint32_t *ptr32 = (uint32_t *)addr;
+	int i;
+
+	rule->data.category_mask = 0x1;
+	rule->data.priority = 1;
+	/* Userdata is filled in in register_ipv6_acl(). */
+
+	rule->field[PROTO_FIELD_IPV6].value.u8 = IPPROTO_ICMPV6;
+	rule->field[PROTO_FIELD_IPV6].mask_range.u8 = 0xFF;
+
+	for (i = DST1_FIELD_IPV6; i <= DST4_FIELD_IPV6; i++) {
+		rule->field[i].value.u32 = rte_be_to_cpu_32(*ptr32);
+		rule->field[i].mask_range.u32 = 32;
+		ptr32++;
+	}
+
+	rule->field[TYPE_FIELD_ICMPV6].value.u32 =
+		(ICMPV6_ECHO_REQUEST_TYPE << 24) & 0xFFFF0000;
+	rule->field[TYPE_FIELD_ICMPV6].mask_range.u32 = 0xFFFF0000;
+}
+
+static int
+register_ping6_acl_rules(struct gatekeeper_if *iface)
+{
+	struct ipv6_acl_rule ipv6_rules[NUM_ACL_PING_RULES];
+	int ret;
+
+	memset(&ipv6_rules, 0, sizeof(ipv6_rules));
+
+	fill_ping6_rule(&ipv6_rules[0], &iface->ip6_addr);
+	fill_ping6_rule(&ipv6_rules[1], &iface->ll_ip6_addr);
+	fill_ping6_rule(&ipv6_rules[2], &iface->ip6_mc_addr);
+	fill_ping6_rule(&ipv6_rules[3], &iface->ll_ip6_mc_addr);
```

Comment:
Does ICMP reply on all these addresses?

Reply:
According to the [RFC 4443](https://tools.ietf.org/html/rfc4443#section-4.1) sections 4.1 and 4.2, the destination address of the ping request message can be any legal IPv6 address.

Reply:
Ok. When one pings the destination `ll_ip6_mc_addr`, what should be the source address of the reply? `ll_ip6_mc_addr`? I'm asking because `ll_ip6_mc_addr` is a "funny" address.

Reply:
Good point!

On section 4.2, it says ``An Echo Reply SHOULD be sent in response to an Echo Request message sent to an IPv6 multicast or anycast address.  In this case, the source address of the reply MUST be a unicast address belonging to the interface on which the Echo Request message was received.`` Maybe we can use the ``iface->ip6_addr`` as the source in this case?

Reply:
Yes. We must use `iface->ip6_addr` as the source address of the reply.

Please also check the expected behavior when the destination is `l_ip6_addr` or `ip6_mc_addr`.